CRM/Core/I18n/Schema.php

```diff
@@ -188,18 +188,20 @@ public static function makeSinglelingualTable(
       }
     }
 
+    $dao = new CRM_Core_DAO();
     // deal with columns
     foreach ($columns[$table] as $column => $type) {
-      $queries[] = "ALTER TABLE {$table} ADD {$column} {$type}";
-      $queries[] = "UPDATE {$table} SET {$column} = {$column}_{$retain}";
+      $queries[] = "ALTER TABLE {$table} CHANGE `{$column}_{$retain}` `{$column}` {$type}";
       foreach ($locales as $loc) {
-        $dropQueries[] = "ALTER TABLE {$table} DROP {$column}_{$loc}";
+        if (strcmp($loc,$retain) !== 0) {
```

Comment:
Since you raised the spectre of MySQL 5.1 in JIRA - I'm not fussed, but easy to account for it?

I thought CiviCRM might already have status available to check but it seems we [query MySQL for this as we need](https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/VersionCheck.php#L268).

    $mysql_version = CRM_CORE_DAO::singleValueQuery('SELECT VERSION()');
    foreach ($locales as $loc) {
      if (version_compare($mysql_version, "5.1")) {
        $queries[] = "DROP VIEW IF EXISTS {$table}_{$loc}";
      }
      else {
        $queries[] = "DROP VIEW {$table}_{$loc}";
      }
    }